include/deal.II/lac/petsc_ts.h

```diff
@@ -0,0 +1,706 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2022 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE.md at
+//    the top level directory of deal.II.
+//
+//---------------------------------------------------------------
+//
+// Author: Stefano Zampini, King Abdullah University of Science and Technology.
+
+#ifndef dealii_petsc_ts_h
+#define dealii_petsc_ts_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_PETSC
+#  include <deal.II/base/mpi.h>
+#  include <deal.II/base/parameter_handler.h>
+#  include <deal.II/base/smartpointer.h>
+
+#  include <deal.II/lac/petsc_matrix_base.h>
+#  include <deal.II/lac/petsc_precondition.h>
+#  include <deal.II/lac/petsc_vector_base.h>
+
+#  include <petscdm.h>
+#  include <petscts.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+// Shorthand notation for PETSc error codes.
+#  define AssertTS(code)                                 \
+    do                                                   \
+      {                                                  \
+        PetscErrorCode __ierr = (code);                  \
```

Comment:
The C++ standard reserves all variables starting with either underscore+Capital letter, or starting with two underscores for compiler and standard library use. You need to use a different name for these variables. (See also #15071.)